### PR TITLE
addressing y-ticks for logscale at barplot

### PR DESCRIPTION
**Pull request by moontrip**
This will address [#185 at web-components](https://github.com/VEuPathDB/web-components/issues/185) and requires [the corresponding PR at web-components](https://github.com/VEuPathDB/web-components/pull/225). 

Here are some examples for both count and proportion cases: used GEMS1A study
case or control participant1

case or control participant2

### CODE REVIEW

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 1]
```diff
@@ -222,6 +222,31 @@ function BarplotViz(props: VisualizationProps) {
...
+      ? Math.max.apply(
+          Math,
+          data?.value?.series.map((o) => {
+            let localMax = Math.max(...o.value);
+            console.log('local max =', localMax);
+            return localMax;
+          })
+        )
+      : undefined;
+  }, [data, data.value, variable, overlayVariable]);
```
Comment (bobular) [inline src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:237]: Don't need both `data` and `data.value` ?

Comment (bobular) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:237]: Elsewhere we've used flatMap and lodash Max, I think?

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx [hunk 2]
```diff
@@ -222,6 +222,31 @@ function BarplotViz(props: VisualizationProps) {
...
+
+  // set min/max
+  const dependentAxisRange =
+    defaultDependentMaxValue != null
+      ? {
+          // set min as 0 (count) or 0.001 (proportion)
+          min: vizConfig.valueSpec === 'count' ? 0 : 0.001,
+          // add 5 % margin
+          max: defaultDependentMaxValue * 1.05,
```
Comment (bobular) [inline src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:246]: nice

File: src/lib/core/utils/default-dependent-axis-range.ts [hunk 3]
```diff
@@ -4,14 +4,14 @@ import { NumberOrDateRange } from '@veupathdb/components/lib/types/general';
...
   plotName: string,
-  // yMinMaxRange: NumberOrDateRange | undefined,
   yMinMaxRange:
     | { min: number | string | undefined; max: number | string | undefined }
     | undefined
 ): NumberOrDateRange | undefined {
   // make universal range variable
-  if (variable != null && variable.dataShape === 'continuous') {
-    if (variable.type === 'number') {
+  if (variable != null && plotName === 'scatterplot') {
```
Comment (bobular) [inline src/lib/core/utils/default-dependent-axis-range.ts:12]: Is this a fix for another ticket?

File: src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx (hunk 1 above)
Comment (moontrip) [reply src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx:237]: @bobular thank you for your review! As for the scatterplot fix, honestly I don't know if it is necessary so your input would be helpful. Here is the situation I have faced for that. When I checked one of my favorite test variables (haha) for scatterplot viz, Sleeping rooms in dwelling (X) & Persons sleeping in dwelling count, I realized that the type of Persons sleeping in dwelling count is now classified as integer (previously number). default-dependent-axis-range.ts only checks number or date so if integer type comes in, no (dependent axis) range is set. I thought that such an integer type should also have the (fixed) range like number. The change is related to this. Do you think this is reasonable to change ?

As for data and data.value, yeah actually one of them works. Perhaps data will be better: will change it

**Pull request merged**